op-node/p2p/gossip.go

```diff
@@ -860,11 +961,46 @@ func verifyBlockResponseSignature(log log.Logger, cfg *rollup.Config, runCfg Gos
 	return pubsub.ValidationAccept
 }
 
+// CHANGE(taiko): verifies the preconfirmation commitment signature
+func verifyPreconfirmationSignature(log log.Logger, cfg *rollup.Config, runCfg GossipRuntimeConfig, id peer.ID, sc *SignedCommitment) pubsub.ValidationResult {
+	// Marshal commitment (without signature)
+	var buf bytes.Buffer
+	if _, err := sc.Commitment.MarshalSSZ(&buf); err != nil {
+		log.Warn("failed to encode preconf commitment for signing", "err", err, "peer", id)
+		return pubsub.ValidationReject
+	}
+
+	signingHash, err := BlockSigningHash(cfg, buf.Bytes())
+	if err != nil {
+		log.Warn("failed to compute preconf signing hash", "err", err, "peer", id)
+		return pubsub.ValidationReject
+	}
+
+	pub, err := crypto.SigToPub(signingHash[:], sc.Signature)
+	if err != nil {
+		log.Warn("invalid preconf signature", "err", err, "peer", id)
+		return pubsub.ValidationReject
+	}
+	addr := crypto.PubkeyToAddress(*pub)
+
+	// If schedule provides a committer, enforce it strictly.
+	if sched, ok := runCfg.(PreconfScheduleRuntime); ok {
+		if committer := sched.CurrentPreconferCommitter(); committer != (common.Address{}) {
+			if addr == committer {
+				return pubsub.ValidationAccept
+			}
+		}
+	}
+	return pubsub.ValidationAccept
```

Comment:
this function always returns accept? i think we need to return reject if the addr != committer?